src/open_inwoner/accounts/views/cases.py

```diff
@@ -88,17 +87,31 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
         cases = fetch_cases(self.request.user.bsn)
-        case_types = cache.get("case_types")
-        if not case_types:
-            case_types = {case_type.url: case_type for case_type in fetch_case_types()}
-            cache.set("case_types", case_types, 60 * 60)
-        status_types = {
-            status_type.url: status_type for status_type in fetch_status_types()
-        }
-        current_statuses = {
-            status.zaak: status
-            for status in fetch_specific_statuses([case.status for case in cases])
-        }
+
+        case_types = {}
+        status_types = {}
+        temp_status_types = []
+        current_statuses = {}
+
+        for case in cases:
+            # Fetch new case types
+            if case.zaaktype not in case_types.keys():
+                case_type = fetch_single_case_type(case.zaaktype)
+                case_types.update({case_type.url: case_type})
```

Comment:
We can use assignment instead if `.update()`, like `case_types[case_type.url] = case_type` (a fewe more below)